modules/javafx.controls/src/main/java/javafx/scene/control/skin/VirtualFlow.java

```diff
@@ -2110,22 +2158,45 @@ boolean addTrailingCells(boolean fillEmptyCells) {
         return filledWithNonEmpty;
     }
 
-    void reconfigureCells() {
+    /**
+     * Informs the {@code VirtualFlow} that a layout pass should be done and cell contents (For example, in the {@code
+     * TableView}, if no cells has been added inside a row) are the same.
+     *
```

Comment:
* "the `{@code VirtualFlow}`" -> "the `{@code VirtualFlow}` container/region/handler/system...", it has to be something.
* "and cell contents" comma before "and" and "and that [the? or any cells?] cell contents"
* "For example" -> "for example"
* "cells has been" -> "cells have been"
* I still don't understand: cell contents are the same as what?

Reply:
I don't really know what the previous developer intended when he used the "cell contents are the same". My deduction base on other methods is that the number of cells inside a row has not changed. 
Don't forget that the VirtualFlow applied to the TableView means that the "cells" we are refering in the VirtualFlow are in fact "rows". So I guess that "cell content are the same" means the numbers of cells inside the row has not changed and we do not need to remove them and re-insert them. But again, I only seen it called at one place and it was pretty confusing.

Reply:
Then we require clarification from someone who is familiar with the implementation. @kevinrushforth, can you help clarify?

Reply:
All of these methods are doing a layout, with some little differences in order to optimise the layout I guess.
Regarding reconfigureCells(), it's only called by the TableViewSkin when its own `needCellsReconfigured` variable is set to true.

This variable is set to true at two places :
```
if (newCount != oldCount) {
            // FIXME updateItemCount is called _a lot_. Perhaps we can make rebuildCells
            // smarter. Imagine if items has one million items added - do we really
            // need to rebuildCells a million times? Maybe this is better now that
            // we do rebuildCells instead of recreateCells.
            requestRebuildCells();
        } else {
            needCellsReconfigured = true;
        }
```
That made me add the comment about the number of cells inside a row.

And 

```
 private InvalidationListener widthListener = observable -> {
        // This forces the horizontal scrollbar to show when the column
        // resizing occurs. It is not ideal, but will work for now.

        // using 'needCellsReconfigured' here rather than 'needCellsRebuilt'
        // as otherwise performance suffers massively (RT-27831)
        needCellsReconfigured = true;
        if (getSkinnable() != null) {
            getSkinnable().requestLayout();
        }
    };
```

If we look inside the VirtualFlow at the impact of this variable :
```
setMaxPrefBreadth(-1);
            lastWidth = -1;
            lastHeight = -1;
```

My guess is that this method is requesting a layout when the Table width has changed and that nothing has changed internally inside the rows. But I'm no expert on the inner behaviour of the VirtualFlow and the subtleties between all these methods.